src/pygenesis/find.py

```diff
@@ -0,0 +1,269 @@
+                        "Name:",
+                        response["Object"]["Content"],
+                        f"{'-' * 20}",
+                        "Content:",
+                        "\n".join(
+                            [content["Content"] for content in axis_dict]
+                        ),
+                        f"{'-' * 40}",
+                    ]
+                )
+
+            elif self.category == "statistics":
+                structure_dict = response["Object"]
+
+                output = "\n".join(
+                    [
+                        f"{self.category.upper()} {code} - {ix}",
+                        "Name:",
+                        response["Object"]["Content"],
+                        f"{'-' * 20}",
+                        "Content:",
+                        "\n".join(
+                            [
+                                f"{structure_dict[content]} {content}"
+                                for content in ["Cubes", "Variables", "Updated"]
+                            ]
+                        ),
+                        f"{'-' * 40}",
+                    ]
+                )
+
+            elif self.category == "variables":
+                object_dict = response["Object"]
+
+                output = "\n".join(
+                    [
+                        f"{self.category.upper()} {code} - {ix}",
+                        "Name:",
+                        object_dict["Content"],
+                        f"{'-' * 20}",
+                        "Information:",
+                        str(object_dict["Information"]),
+                        f"{'-' * 40}",
+                    ]
+                )
+
+            print(output)
+
+    @staticmethod
+    def _get_metadata_results(category: str, code: str):
+        """
+        Based on the category and code query parameters the metadata will be generated.
+
+        Args:
+            category (str): Category of the result. E.g. "tables", "cubes"
+            code (str): The code (identifier) of the relevant category object.
+        Returns:
+            response (json): The response as a json.
+        """
+        params = {
+            "name": code,
+        }
+
+        response = get_response_from_endpoint("metadata", category, params)
+        response_json = response.json()
+
+        return response_json
+
+
+class Find:
+    """
+    A class representing the find object that includes Result objects for variables, statistics, cubes and tables.
+
+    Attributes:
+        query (str): The query that is provided to find endpoint.
+        top_n_preview (int): Number of previews in print summary.
+
+    Methods:
+        _print_summary(): Prints a summary with result previews.
+        _get_find_results(): Gets results from the find endpoints.
+    """
+
+    def __init__(self, query: str, top_n_preview: int = 5) -> None:
+        """Method for retrieving data from find endpoint.
```

Comment:
Ich finde es etwas unschön, dass in der init direkt die suche gestartet wird, ich persönlich würde es favorisieren, wenn man das Objekt erst initialisiert mit einer query, und sich dann bewusst dafür entscheiden kan ndie Suche zu starten. Ansonsten hat man immer eine große Wartezeit wenn man das Objekt nur anlegen oder initialisern will. Also die Aufrufe von `_get_find_results` sollten außerhalb der __init__ sein. Ich empfehle dafür eine `.run()` Methode oder so